src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -157,7 +157,19 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
                 }
                 else
                 {
-                    content.Add(generator.Interpolation(piece.WithoutTrivia()));
+                    var firstInterpolationToken = piece.GetFirstToken();
+                    var lastInterpolationToken = piece.GetLastToken();
+
+                    if (firstInterpolationToken.RawKind == startToken.RawKind && lastInterpolationToken.RawKind == endToken.RawKind)
+                    {
+                        var previousIterpolatedText = piece.ToString();
+                        var textWithoutInterpolationTokens = previousIterpolatedText.Replace("$", "").Replace("\"", "");
+                        content.Add(generator.InterpolatedStringText(generator.InterpolatedStringTextToken(textWithoutInterpolationTokens)));
```

Comment:
i'm not really understanding what's going on here.  can you doc the code (with a running example) to explain why the above is correct?  Thanks!

Reply:
ok

Reply:
i made it

Reply:
The `.Replace` is a problem if there is a `$` inside the string. There needs to be a test of what happens when `piece` is `$"$ {a} \""`.

Reply:
oh i see 
this can be solve by using substring of the piece 
like : Substring(2, ((pieceString.Length - 1) - 2))

Reply:
oh i see 
this can be solve by using substring of the piece 
like : Substring(2, ((pieceString.Length - 1) - 2))

Reply:
can you add some more tests here as well? i'm not really convinced that we have enough testing to validate that the above is correct.  thanks!

Reply:
I added one more test case